lib/cosmos/packets/structure.rb

```diff
@@ -312,7 +312,9 @@ def formatted(value_type = :RAW, indent = 0, buffer = @buffer)
       string = ''
       synchronize_allow_reads(true) do
         @sorted_items.each do |item|
-          if item.data_type != :BLOCK
+          if (item.data_type != :BLOCK) ||
+             (item.data_type == :BLOCK and value_type != :RAW and
+              item.respond_to? :read_conversion and item.read_conversion)
```

Comment:
Initially I thought you didn't need this respond_to? but since only packet_items have read_conversions and structure_items do not this is necessary. Correct?